plane/src/drone/runtime/docker/mod.rs

```diff
@@ -194,6 +194,37 @@ impl Runtime for DockerRuntime {
     async fn terminate(&self, backend_id: &BackendName, hard: bool) -> Result<bool, anyhow::Error> {
         let container_id: ContainerId = backend_id.into();
 
+        // check if container is no longer running, since stop_container() returns Ok(()) even when the container is already gone
+        match self
+            .docker
+            .inspect_container(&container_id.to_string(), None)
+            .await
+        {
+            Ok(details) => {
+                if let Some(state) = details.state {
+                    if !state.running.unwrap_or(false) {
+                        tracing::warn!(
+                            %container_id,
+                            %backend_id,
+                            "Container could not be terminated, because it is not running."
+                        );
+                        return Ok(false);
+                    }
+                }
```

Comment:
I'm not a fan of nested `if` or `if lets`, find this hard to read. 

How about instead:
```Rust
if !details.state.and_then(|s| s.running).unwrap_or(false) {
    tracing::warn!(
        %container_id,
        %backend_id,
        "Container could not be terminated, because it is not running."
    );
    return Ok(false);
}
```

Reply:
Updated!